guardian/gateway
# MVP1 Design Changes

## What does this change?

Replaces the old layout/designs with a new layout with the template and designs defined in the [figma file](https://www.figma.com/file/ZDPonnBXbMhV2smf51Dylc/?node-id=3628%3A102523), and updates all the pages that used the old layout (`Main`) to use the new one (`MainLayout`).

Most of the design work was centered around the [`MainLayout`](https://github.com/guardian/gateway/blob/mm/mvp1-design/src/client/layouts/Main.tsx) component, which is used to structure a given page with the new design. It defines the grid layout, spacings, and has props to set the title/heading, as well as any success or error messages using the new `Summary` components. Content that goes into this layout is done using `children`.

Alongside the `MainLayout`, some helper components were created to simplify how to layout the `children` to the layout.
Specifically the `MainBodyText` and `MainForm` (+ related `PasswordFormMainLayout`) components.

[`MainBodyText`](https://github.com/guardian/gateway/blob/mm/mvp1-design/src/client/components/MainBodyText.tsx) is the component to use to add text/paragraphs corresponding to the designs in this layout.

[`MainForm`](https://github.com/guardian/gateway/blob/mm/mvp1-design/src/client/components/MainForm.tsx) is used to wrap all forms that are used on the pages using the `MainLayout` and helps to simplify form creation. By default it includes the `CsrfFormField`, a customizable submit button (through props), as well as the optional ability to add the terms and conditions for the Guardian and Recaptcha separately. Input fields are passed as `children` to this component, and will automatically be picked up as a form field. At the very least a `formAction` (url to `POST` the form to), and `submitButtonText` must be provided.
In the future when implementing recaptcha, the `useRecaptcha` flag could be used to easily enable/disable recaptcha for a given form on a page.

Once those components (and respective stories) were in place much of the rest of the diffs in this PR is the refactoring of all of the pages from the old layout to the new layout.

One thing for the future is to combine the `MainGrid` layout (used for the sign in/register pages) with the new `MainLayout` component now used for many of the other pages.

## How to test/review this PR

- Open the [figma file](https://www.figma.com/file/ZDPonnBXbMhV2smf51Dylc/?node-id=3628%3A102523) on the `✅ -  Check your inbox - refinement` page to see expected designs
- Open the [latest storybook](https://60929d168594f80039336501-plgtaqvjar.chromatic.com/) for this branch to see new designs
- Open the [latest chromatic UI tests](https://www.chromatic.com/build?appId=60929d168594f80039336501&number=1416) to see what's changed
